src/main/java/seedu/address/logic/commands/LessonEditCommand.java

```diff
@@ -143,7 +144,9 @@ public Lesson createEditedLesson(Lesson lessonToEdit, EditLessonDescriptor editL
             throws CommandException {
         assert lessonToEdit != null;
         Date updatedDate = editLessonDescriptor.getDate().orElse(lessonToEdit.getStartDate());
-        Date updatedEndDate = editLessonDescriptor.getEndDate().orElse(lessonToEdit.getEndDate());
+        Date updatedEndDate = lessonToEdit.isRecurring()
+                ? editLessonDescriptor.getEndDate().orElse(lessonToEdit.getEndDate())
+                : updatedDate;
```

Comment:
Code for bug fix

Reply:
Rest is adding clashing lessons message 

Tests are for ladd, ledit with end date